File: apps/mobile/src/types/product.ts
export interface Product {
  id: string;
  name: string;
  brand: string;
  price: number;
  original_price?: number;
  currency: string;
  image_url: string;
  product_url: string;
  category?: string;
  is_on_sale?: boolean;
  description: string;
  sizes?: string[];
  colors?: string[];
  material?: string;
  gender_tag?: string;
  source: string;
  created_at: string;
}

export interface UserProductInteraction {
  user_id: string;
  product_id: string;
  interaction_type: 'bought' | 'favorited' | 'viewed';
  created_at: string;
}

export type ProductGridType = 'closet' | 'wishlist' | 'recommended';

File: apps/api/app/models/product.py
import uuid
from sqlalchemy import Column, String, Float, Boolean, Text, ARRAY, DateTime
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.ext.declarative import declarative_base
from sqlalchemy.sql import func
from pgvector.sqlalchemy import Vector

Base = declarative_base()

class Product(Base):
    __tablename__ = 'products'

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    name = Column(String, nullable=False)
    brand = Column(String, nullable=True)
    price = Column(Float, nullable=False)
    original_price = Column(Float, nullable=True)
    currency = Column(String, nullable=False)
    image_url = Column(String, nullable=False)
    product_url = Column(String, nullable=False, unique=True)
    category = Column(String, nullable=True)
    is_on_sale = Column(Boolean, default=False)
    description = Column(Text, nullable=True)
    sizes = Column(ARRAY(String), nullable=True)
    colors = Column(ARRAY(String), nullable=True)
    material = Column(String, nullable=True)
    gender_tag = Column(String, nullable=True)
    source = Column(String, nullable=False)
    embedding = Column(Vector(768)) # 768 dimensions  
    created_at = Column(DateTime(timezone=True), server_default=func.now()) 